lm/srilm.py

```diff
@@ -94,11 +95,103 @@ def hash(cls, kwargs):
         return super().hash(kwargs)
 
 
+class DiscountNgramsJob(Job):
+    """
+    Create a file with the discounted ngrams with SRILM
+    """
+
+    def __init__(
+        self,
+        ngram_order: int,
+        counts: tk.Path,
+        count_exe: tk.Path,
+        *,
+        vocab: Optional[tk.Path] = None,
+        extra_discount_args: Optional[List[str]] = None,
+        use_modified_srilm: bool = False,
+        cpu_rqmt: int = 1,
+        mem_rqmt: int = 48,
+        time_rqmt: float = 24,
+        fs_rqmt: str = "100G",
+    ):
+        """
+        :param ngram_order: order of the ngram counts, typically 3 or 4.
+        :param counts: file with the ngram counts, see :class:`CountNgramsJob.out_counts`.
+        :param vocab: vocabulary file for the discounting.
+        :param count_exe: path to the binary.
+        :param extra_discount_args: additional arguments for the discounting step.
+        :param use_modified_srilm: Use the i6 modified SRILM version by Sundermeyer.
+                                   The SRILM binary ngram-count was modified.
+                                   This version is currently only available internally.
+        :param cpu_rqmt: CPU requirements.
+        :param mem_rqmt: memory requirements.
+        :param time_rqmt: time requirements.
+        :param fs_rqmt: file size requirements.
+        """
+        self.ngram_order = ngram_order
+        self.counts = counts
+        self.vocab = vocab
+        self.discount_args = extra_discount_args or []
+        self.use_modified_srilm = use_modified_srilm
+
+        self.count_exe = count_exe
+
+        self.out_discounts = self.output_path("discounts", cached=True)
+
+        self.rqmt_run = {
+            "cpu": cpu_rqmt,
+            "mem": mem_rqmt,
+            "time": time_rqmt,
+            "qsub_args": f"-l h_fsize={fs_rqmt}",
```

Comment:
Should this really have SGE specific things hardcoded in it? I think it would be better to add them outside the job if they're required.

Reply:
I removed the SGE specific rqmts from the job I added. I am not sure about the other jobs since that might affect other setups...